system/security/LdapSecurity/ldapsecurity.cpp

```diff
@@ -667,22 +686,41 @@ bool CLdapSecManager::authenticate(ISecUser* user)
     }
 
     //Verify provided signature if present
-    IDigitalSignatureManager * pDSM = queryDigitalSignatureManagerInstanceFromEnv();
-    if (pDSM && pDSM->isDigiVerifierConfigured() && !isEmptyString(user->credentials().getSignature()))
+    IDigitalSignatureManager * pDSM = nullptr;
+    if (!isEmptyString(user->credentials().getSignature()))
     {
-        StringBuffer b64Signature(user->credentials().getSignature());
-        if (!pDSM->digiVerify(user->getName(), b64Signature))//digital signature valid?
+        if (isUserCached)
         {
-            user->setAuthenticateStatus(AS_INVALID_CREDENTIALS);
-            WARNLOG("Invalid digital signature for user %s", user->getName());
-            return false;
+            if (streq(cachedUser->credentials().getSignature(), user->credentials().getSignature()))
```

Comment:
could signature have been null last time when it was cached?

Reply:
I see no obvious code path where this is possible and did not come across it in my testing